Card/ Explore Housing Policy in the Portland Area

jaronheard opened the pull request:
This implements #106, with a simplified design that I worked on with Cat. Some interesting information hierarchy challenges in this one.

![image](https://user-images.githubusercontent.com/7065695/45660670-cced9c00-baae-11e8-9caa-7ba7991a0037.png)

DingoEatingFuzz commented:
Got some questions.

**1. What do the legend markers mean?**

The idea of a legend is great and really helpful for skimming through the list, but I have no idea what low/medium/high means in this context. Is it a measure of commonality? If so, I would use words like rare/uncommon/common. Is it a measure of complexity to implement or enforce? If so, I would use words like simple/typical/difficult. 

I would also add a sentence explanation because even more specific word choice in the legend leaves room for (mis)interpretation.

**2. Do we expect users to be generally interested in policy, or their personal policies?**

I recall some of the original designs optimizing for the "what are the policies in _my_ municipality?" I know this design still has that information, but it is buried. 

**3. Can we do something beyond links for the headings?**

This strikes me as a typical accordion display, but it is masquerading as a list. I think if the clicking items had a bottom border instead of a text decoration it would go a long way towards giving the information structure.

**4. What do you think about adding a caveats/addendums subsection?**

Right now the caveats for each implementing government is listed inline with the list of governments. This makes it even harder to skim the list of implementing governments. If we broke it into two sections, then the list of governments could be quickly scanned and the details for any government with caveats could be read below. Like:

--------------------------

**Implementing governments**

  - City of Beaverton
  - City of Hillsboro
  - Multnomah County
  - City of Portland
  - State of Oregon
  - City of Tigard
  - Washington County

**Caveats by government**

  - City of Hillsboro
    - None. Currently exploring nonprofit-based options. No MULTE.
  - City of Portland
    - Multiple-Unit Limited Tax Exemption Program: Program provides a ten-year property tax exemption on affordable residential housing developments with 20 or more units.
      - City of Portland: Multiple-Unit Limited Tax Exemption (MULTE) Overview ()
  - State of Oregon
    - Covered by Vertical Housing Zones and various grant programs.
  - City of Tigard
    - Nonprofit Corporation Low Income Housing Tax Exemption: Properties approved for the exemption can utilize tax savings for operating costs, and save developer fees for future housing projects.
      - Tigard Affordable Housing Strategies 2002-2016 ()
  - Washington County
    - Multifamily Tax Exemption: Tax exemption for nonprofit multifamily developers that rent housing at 50% AMI. Can get a property tax exemption.
      - Washington County Housing Services ()

-----------------------------

Aside from that bike-shedding, great job tackling this hard info problem. It's always nice seeing new patterns being added to the platform.

jaronheard commented:
1. What do the legend markers mean?
Right now, it doesn't represent something that is especially meaningful, as it is counting the individual policies - not the implementing governments. I think what is potentially meaningful is percentage of governments implementing the policy. I'm thinking that using the common / uncommon / rare (mythic rare?) structure will work.

2. Do we expect users to be generally interested in policy, or their personal policies?
This design is optimized for organization by policy - with the implicit expectation that users will be interested in policy. I would be interested in exploring how to incorporate that extra dimension, but I don't have a great concept for how it could work.

3. Can we do something beyond links for the headings?
I like the idea of using a bottom border, especially with the removal of link underlining per the style guide and #392. I'll try a few things.

4. Do you think it would help to add a count of implementing governments to each heading?
I think that could be a nice addition. Honestly, something close to this GitHub design, with colored icon, large text for policy name, and smaller text for governments implementing, and some separation between each heading could work nicely.
![image](https://user-images.githubusercontent.com/7065695/45779137-1f45ce80-bc0f-11e8-8e09-cba381308627.png)

5. What do you think about adding a caveats/addendums subsection?
I love it. Big improvement.

jaronheard commented:
I did a redesign of this based on @DingoEatingFuzz's comments.

Updated design:
![image](https://user-images.githubusercontent.com/7065695/45865608-ebfc6000-bd32-11e8-9923-7a9c261cd630.png)

Expanded:
![image](https://user-images.githubusercontent.com/7065695/45865570-d1c28200-bd32-11e8-8df7-413e507008cd.png)

Feeling pretty good about the new design, but not sure that what I'm showing in terms of common/uncommon/rare implementation is the best possible. I also think the code could use another look.

DingoEatingFuzz commented on packages/2018-housing-affordability/src/components/ExploreHousingPolicyImplementation/index.js:
```diff
@@ -1,43 +1,179 @@
+          In the spring and summer of 2018,{' '}
+          <a href="http://www.hackoregon.org/">Hack Oregon</a> volunteers
+          surveyed local governments to understand what policies were being
+          implemented. Taking inspiration from the{' '}
```
What a cool jsx quirk.

jaronheard replied:
Been using prettify and didn't actually realize what was going on at first - I guess it's just so that you can see the whitespace when you're reading the code.

DingoEatingFuzz replied:
It's more nefarious than that. [JSX strips all newlines between elements](https://github.com/facebook/react/issues/1643). This is contrary to HTML which treats newlines as whitespace characters. So Prettier must be protecting you from losing the expected space between `the` and the following `a` element.

DingoEatingFuzz commented on packages/2018-housing-affordability/src/state/housing-policy/selectors.js:
```diff
@@ -0,0 +1,96 @@
+
+    const selectedPolicyData = policies.filter(
+      policy => policy.policy_id === selectedPolicy
+    );
```
You could use `policies.find` here instead. Then you won't have to do all the `selectedPolicyData[0]` stuff below.

jaronheard commented:
@DingoEatingFuzz Thanks for the comments. I refactored the confusing code and incorporated the other suggestions, so I think this should be ready to go.
